### PR TITLE
Update react nextjs

### PR DESCRIPTION
**Pull request by Ruilin-Ma**

### CODE REVIEW

**Comment by gkwan-ibm** [comment]
- [x] in pom.xml, under the `maven-resources-plugin`, format the content in between the  `<executions>` and `</executions>` tags better 
- **Creating the default page**
  - [x] "The `page.jsx` file will be act ..."  -> "The `page.jsx` file will be acted ..."  or "The `page.jsx` file will act ..." 
  - [x] "...documentation on [Pages and Layouts](https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts)", is it better to link Pages and Layouts to different docs?
- **Building and packaging the front-end**
  - [x] after run `mvn process-resources`, the `http://localhost:9080/` does not work. After press `r` in dev mode console to restart the app, the link work
- not sure which terms are correct: `front end`/`front-end`/`frontend` and  `back end`/`back-end`/`backend`
  - [ ] need ID to confirm and update the guide to use them consistently

File: README.adoc [hunk 1]
```diff
@@ -174,73 +174,69 @@ include::finish/pom.xml[]
...
+In order to render the pages correctly, it is required to add a [hotspot file=1]`layout.jsx` file which defines the `RootLayout` containing the UI that shared across all routes.
+
+[role="code_command hotspot file=1", subs="quotes"]
+----
+#Create the `layout.jsx` file.#
+`src/main/frontend/src/app/layout.jsx`
+----
 
-The [hotspot=dom-render file=0]`render` method takes an HTML DOM element and tells the ReactDOM to render your React application inside of this DOM element. To learn more about the React virtual DOM, see the https://reactjs.org/docs/react-dom.html[ReactDOM^] documentation.
+For more detailed information on structuring `Pages` and `Layouts` in `Next.js`, please refer to the `Next.js` documentation on https://nextjs.org/docs/app/building-your-application/routing/pages[Pages] and https://nextjs.org/docs/app/building-your-application/routing/layouts-and-templates[Layouts]
```
Comment (gkwan-ibm) [inline README.adoc]: missed a period at the end of the statement

File: README.adoc [hunk 2]
```diff
@@ -174,73 +174,69 @@ include::finish/pom.xml[]
...
 
 // =================================================================================================
-// Creating the React components
+// Creating ArtistTable React component
 // =================================================================================================
 
-== Creating the React components
+== Creating ArtistTable React component
```
Comment (gkwan-ibm) [inline README.adoc]: no need to change the section title as same as the comment at line 212

File: README.adoc [hunk 3]
```diff
@@ -320,13 +316,12 @@ endif::[]
 
 == Testing the React client
 
-New projects that are created with `create-react-app` comes with a test file called [hotspot file]`App.test.js`, which is included in the `src/main/frontend/src` directory. The `App.test.js` file is a simple JavaScript file that tests against the `App.js` component. There are no explicit test cases that are written for this application. The `create-react-app` configuration uses https://jestjs.io/[Jest^] as its test runner.
-To learn more about Jest, go to their documentation on https://jestjs.io/docs/en/tutorial-react[Testing React apps^]. 
+`Next.js` supports a variety of testing tools. This guide uses `Vitest` for unit testing the React components, with the test file [hotspot file]`App.test.jsx` located in `src/main/frontend/__tests__/` directory. The `App.test.jsx` file is a simple JavaScript file that tests against the `page.jsx` component. There are no explicit test cases that are written for this application. To learn more about `Vitest`, please visit https://nextjs.org/docs/app/building-your-application/testing/vitest[Setting up Vitest with Next.js]
```
Comment (gkwan-ibm) [inline README.adoc]: miss a period at the end

File: README.adoc [hunk 4]
```diff
@@ -35,15 +35,15 @@ You will learn how to access a REST service and deserialize the returned JSON th
...
 
-The React application in this guide is provided and configured for you in the `src/main/frontend` directory. The application uses the https://reactjs.org/docs/create-a-new-react-app.html[Create React App^] prebuilt configuration to set up the modern single-page React application. The https://github.com/facebook/create-react-app[create-react-app^] integrated toolchain is a comfortable environment for learning React and is the best way to start building a new single-page application with React.
+The React application in this guide is provided and configured for you in the `src/main/frontend` directory. The application uses https://nextjs.org/[Next.js], one of the https://react.dev/learn/start-a-new-react-project[React-powered frameworks], to set up the modern React application. The `Next.js` framework provides a powerful environment for learning and building React applications, with features like server-side rendering, static site generation, and easy API routes. It is the best way to start building a new, highly performant React application.
```
Comment (gkwan-ibm) [inline README.adoc]: - miss the `^` char in the links
  - `[Next.js]` -> `[Next.js^]`
  - `[React-powered frameworks]` -> `[React-powered frameworks^]`

**Comment by gkwan-ibm** [comment]
- for `sed -i 's=http://localhost:9080/artists='"https://${USERNAME}-9080.$(echo $TOOL_DOMAIN | sed 's/\.labs\./.proxy./g')/artists"'=' src/main/webapp/static/js/main.2d7e902e.js` 
  - update `main.2d7e902e.js` to `main.1bcb490d.js`

**Review by shin19991207 — APPROVED** [review]
Tested on Skills Network and Windows, no issues.

File: README.adoc [hunk 5]
```diff
@@ -35,15 +35,15 @@ You will learn how to access a REST service and deserialize the returned JSON th
...
 
-The React application in this guide is provided and configured for you in the `src/main/frontend` directory. The application uses the https://reactjs.org/docs/create-a-new-react-app.html[Create React App^] prebuilt configuration to set up the modern single-page React application. The https://github.com/facebook/create-react-app[create-react-app^] integrated toolchain is a comfortable environment for learning React and is the best way to start building a new single-page application with React.
+The React application in this guide is provided and configured for you in the `src/main/frontend` directory. The application uses https://nextjs.org/[Next.js^], one of the https://react.dev/learn/start-a-new-react-project[React-powered frameworks^], to set up the modern React application. The `Next.js` framework provides a powerful environment for learning and building React applications, with features like server-side rendering, static site generation, and easy API routes. It is the best way to start building a new, highly performant React application.
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
The React application in this guide is provided and configured for you in the `src/main/frontend` directory. The application uses https://nextjs.org/[Next.js^], a https://react.dev/learn/start-a-new-react-project[React-powered frameworks^], to set up the modern React application. The `Next.js` framework provides a powerful environment for learning and building React applications, with features like server-side rendering, static site generation, and easy API routes. It is the best way to start building a highly performant React application.
```

File: README.adoc [hunk 6]
```diff
@@ -35,15 +35,15 @@ You will learn how to access a REST service and deserialize the returned JSON th
...
 
 artists.json
 [source, json, linenums, role="code_column"]
 ----
 include::finish/src/resources/artists.json[]
 ----
 
-The REST service that provides the resources was written for you in advance in the back end of the application, and it responds with the [hotspot]`artists.json` in the `src/resources` directory. You will implement a ReactJS client as the front end of your application, which consumes this JSON file and displays its contents on a single-page webpage. 
+The REST service that provides the resources was written for you in advance in the back-end of the application, and it responds with the [hotspot]`artists.json` in the `src/resources` directory. You will implement a ReactJS client as the front-end of your application, which consumes this JSON file and displays its contents on a single-page webpage.
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
The REST service that provides the resources was written for you in advance in the back end of the application, and it responds with the [hotspot]`artists.json` file  in the `src/resources` directory. You will implement a ReactJS client as the front end of your application, which consumes this JSON file and displays its contents on a single web page. 
```

File: README.adoc [hunk 7]
```diff
@@ -59,7 +59,7 @@ include::{common-includes}/gitclone.adoc[]
 
 === Try what you'll build
 
-The `finish` directory in the root of this guide contains the finished application. The React front end is already pre-built for you and the static files from the production build can be found in the `src/main/webapp/static` directory.
+The `finish` directory in the root of this guide contains the finished application. The React front-end is already pre-built for you and the static files from the production build can be found in the `src/main/webapp/_next/static` directory.
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
The `finish` directory in the root of this guide contains the finished application. The React front end is already pre-built for you and the static files from the production build can be found in the `src/main/webapp/_next/static` directory.
```

File: README.adoc [hunk 8]
```diff
@@ -140,15 +139,15 @@ curl -s http://localhost:9080/artists | jq
 ```
 endif::[]
 
-All the dependencies for the React front end can be found in `src/main/frontend/src/package.json`, and they are installed before the front end is built by the `frontend-maven-plugin`. Additionally, some provided `CSS` stylesheets files are provided and can be found in the `src/main/frontend/src/Styles` directory.
+All the dependencies for the React front-end are listed in `src/main/frontend/src/package.json` and are installed before the build process by the `frontend-maven-plugin`. Additionally, `CSS` stylesheets files are available in the `src/main/frontend/src/styles` directory.
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
All the dependencies for the React front end are listed in the  `src/main/frontend/src/package.json` file and are installed before the build process by the `frontend-maven-plugin`. Also, `CSS` stylesheets files are available in the `src/main/frontend/src/styles` directory.
```

File: README.adoc [hunk 9]
```diff
@@ -140,15 +139,15 @@ curl -s http://localhost:9080/artists | jq
...
 
 // =================================================================================================
 // Project configuration 
 // =================================================================================================
 
 == Project configuration
 
-The front end of your application uses Node.js to build your React code. The Maven project is configured for you to install Node.js and produce the production files, which are copied to the web content of your application.
+The front-end of your application uses Node.js to build your React code. The Maven project is configured for you to install Node.js and produce the production files, which are copied to the web content of your application.
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
The front end of your application uses Node.js to build your React code. The Maven project is configured for you to install Node.js and produce the production files, which are copied to the web content of your application.
```

File: README.adoc [hunk 10]
```diff
@@ -174,73 +174,69 @@ include::finish/pom.xml[]
 
 == Creating the default page
 
-You need to create the entry point of your React application. `create-react-app` uses the [hotspot file=0]`index.js` file as the main entry point of the application. This JavaScript file corresponds with the [hotspot file=1]`index.html` file, which is the entry point where your code runs in the browser.
+You'll need to create the entry point of your React application. In the latest version of `Next.js`, it's recommended to use the https://nextjs.org/docs/app/building-your-application/routing/defining-routes[App Router], which centralizes routing logic under the `app` directory.
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
Create the entry point of your React application. The latest version of `Next.js` recommends  you use the https://nextjs.org/docs/app/building-your-application/routing/defining-routes[App Router], which centralizes routing logic under the `app` directory. 
```

File: README.adoc [hunk 11]
```diff
@@ -174,73 +174,69 @@ include::finish/pom.xml[]
...
+page.jsx
 [source, javascript, linenums, role='code_column']
 ----
-include::finish/src/main/frontend/src/index.js[]
+include::finish/src/main/frontend/src/app/page.jsx[]
 ----
 
-index.html
-[source, html, linenums, role='code_column tags=html hide_tags=copyright']
+The `page.jsx` file will serve as a container for all other components. When React component [hotspot=createHome file=0]`Home` is rendered, the [hotspot=displayArtistTable file=0]`ArtistTable` components content will be displayed.
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
The `page.jsx` file is a container for all other components. When the [hotspot=createHome file=0]`Home` React component  is rendered, the [hotspot=displayArtistTable file=0]`ArtistTable` components content are displayed.
```

File: README.adoc [hunk 12]
```diff
@@ -174,73 +174,69 @@ include::finish/pom.xml[]
...
+
+In order to render the pages correctly, it is required to add a [hotspot file=1]`layout.jsx` file which defines the `RootLayout` containing the UI that shared across all routes.
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
To render the pages correctly, add a [hotspot file=1]`layout.jsx` file that defines the `RootLayout` containing the UI that are shared across all routes.
```

File: README.adoc [hunk 13]
```diff
@@ -174,73 +174,69 @@ include::finish/pom.xml[]
...
+layout.jsx
+[source, javascript, linenums, role='code_column']
+----
+include::finish/src/main/frontend/src/app/layout.jsx[]
+----
 
-The [hotspot=dom-render file=0]`render` method takes an HTML DOM element and tells the ReactDOM to render your React application inside of this DOM element. To learn more about the React virtual DOM, see the https://reactjs.org/docs/react-dom.html[ReactDOM^] documentation.
+For more detailed information on structuring `Pages` and `Layouts` in `Next.js`, please refer to the `Next.js` documentation on https://nextjs.org/docs/app/building-your-application/routing/pages[Pages^] and https://nextjs.org/docs/app/building-your-application/routing/layouts-and-templates[Layouts^].
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
For more detailed information, see the `Next.js` documentation on https://nextjs.org/docs/app/building-your-application/routing/pages[Pages^] and https://nextjs.org/docs/app/building-your-application/routing/layouts-and-templates[Layouts^].
```

File: README.adoc [hunk 14]
```diff
@@ -174,73 +174,69 @@ include::finish/pom.xml[]
...
 // =================================================================================================
-// Creating the React components
+// Creating the React component
 // =================================================================================================
 
-== Creating the React components
+== Creating the React component
 
-A React web application is a collection of components, and each component has a specific function. You will create the components that are used in the application to acquire and display data from the REST API. 
+A React web application is a collection of components, and each component has a specific function. You will create a component that are used in the application to acquire and display data from the REST API.
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
A React web application is a collection of components, and each component has a specific function. You will create a component that the application uses to acquire and display data from the REST API. 
```

File: README.adoc [hunk 15]
```diff
@@ -174,73 +174,69 @@ include::finish/pom.xml[]
...
-ArtistTable.js
+ArtistTable.jsx
 [source, javascript, linenums, role='code_column hide_tags=get-posts,useEffect,axios-library']
 ----
-include::finish/src/main/frontend/src/Components/ArtistTable.js[]
+include::finish/src/main/frontend/src/app/ArtistTable.jsx[]
 ----
 
-The [hotspot=react-library file=1]`React` library imports the `react` package for you to create the [hotspot=ArtistTable file=1]`ArtistTable` function. This function must have the `export` declaration because it is being exported to the `App.js` module. The [hotspot=posts file=1]`posts` object is initialized using a React Hook that lets you add a state to represent the state of the posts that appear on the paginated table.
+At the top of the file, the [hotspot=render-client file=0]`use client` directive is used to indicate the `ArtistTable` component will be rendered on the client side.
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
At the beginning of the file, the [hotspot=render-client file=0]`use client` directive indicates the `ArtistTable` component is rendered on the client side.
```

File: README.adoc [hunk 16]
```diff
@@ -174,73 +174,69 @@ include::finish/pom.xml[]
...
 
-The [hotspot=useTable file=1]`useTable` hook creates a paginated table. The `useTable` hook takes in the [hotspot=table-info file=1]`columns`, [hotspot=posts file=1]`posts`, and `setPosts` objects as parameters. It returns a paginated table that is assigned to the [hotspot=table file=1]`table` constant. The [hotspot=table file=1]`table` constant renders the paginated table on the webpage. The [hotspot=return-table file=1]`return` statement returns the paginated table. The [hotspot=useSortBy file=1]`useSortBy` hook sorts the paginated table by the column headers. The [hotspot=usePagination file=1]`usePagination` hook creates the pagination buttons at the bottom of the table that are used to navigate through the paginated table.
+To display the returned data, you will use pagination. Pagination is the process of separating content into discrete pages, and it can be used for handling data sets in React. In your application, you'll render the columns in the paginated table. The [hotspot=table-info file=0]`columns` constant is used to define the table that is present on the webpage.
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
To display the returned data, you will use pagination. Pagination is the process of separating content into discrete pages, and you can use it for handling data sets in React. In your application, you'll render the columns in the paginated table. The [hotspot=table-info file=0]`columns` constant defines the table that is present on the web page.
```

File: README.adoc [hunk 17]
```diff
@@ -174,73 +174,69 @@ include::finish/pom.xml[]
...
+
+The [hotspot=useReactTable file=0]`useReactTable` hook creates a table instance. The hook takes in the [hotspot=table-info file=0]`columns`, [hotspot=posts file=0]`posts` as parameters. The [hotspot=getCoreRowModel file=0]`getCoreRowModel` function is included for the generation of the core row model of the table, which serves as the foundational row model upon pagination and sorting build. The [hotspot=getPaginationRowModel file=0]`getPaginationRowModel` function applies pagination to the core row model, returning a row model that includes only the rows that should be displayed on the current page based on the pagination state. In addition, the [hotspot=getSortedRowModel file=0]`getSortedRowModel` function sorts the paginated table by the column headers then applies the changes to the row model. The paginated table instance is assigned to the [hotspot=table file=0]`table` constant, which renders the paginated table on the webpage.
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
The [hotspot=useReactTable file=0]`useReactTable` hook creates a table instance. The hook takes in the [hotspot=table-info file=0]`columns` and  [hotspot=posts file=0]`posts` as parameters. The [hotspot=getCoreRowModel file=0]`getCoreRowModel` function is included for the generation of the core row model of the table, which serves as the foundational row model upon pagination and sorting build. The [hotspot=getPaginationRowModel file=0]`getPaginationRowModel` function applies pagination to the core row model, returning a row model that includes only the rows that should be displayed on the current page based on the pagination state. In addition, the [hotspot=getSortedRowModel file=0]`getSortedRowModel` function sorts the paginated table by the column headers then applies the changes to the row model. The paginated table instance is assigned to the [hotspot=table file=0]`table` constant, which renders the paginated table on the web page.
```

File: README.adoc [hunk 18]
```diff
@@ -250,18 +246,18 @@ The [hotspot=useTable file=1]`useTable` hook creates a paginated table. The `use
 
 Your application needs a way to communicate with and retrieve resources from RESTful web services to output the resources onto the paginated table. The https://github.com/axios/axios[Axios^] library will provide you with an HTTP client. This client is used to make HTTP requests to external resources. Axios is a promise-based HTTP client that can send asynchronous requests to REST endpoints. To learn more about the Axios library and its HTTP client, see the https://www.npmjs.com/package/axios[Axios documentation^].
 
-The [hotspot=get-posts]`GetArtistsInfo()` function uses the Axios API to fetch data from your back end. This function is called when the `ArtistTable` is rendered to the page using the [hotspot=useEffect]`useEffect()` React lifecycle method.
+The [hotspot=get-posts]`GetArtistsInfo()` function uses the Axios API to fetch data from your back-end. This function is called when the `ArtistTable` is rendered to the page using the [hotspot=useEffect]`useEffect()` React lifecycle method.
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
The [hotspot=get-posts]`GetArtistsInfo()` function uses the Axios API to fetch data from your back end. This function is called when the `ArtistTable` is rendered to the page using the [hotspot=useEffect]`useEffect()` React lifecycle method.
```

File: README.adoc [hunk 19]
```diff
@@ -270,21 +266,21 @@ Add the [hotspot=axios-library]`axios` library and the [hotspot=get-posts]`GetAr
...
 // =================================================================================================
-// Building and packaging the front end
+// Building and packaging the front-end
 // =================================================================================================
 
-== Building and packaging the front end
+== Building and packaging the front-end
 
-After you successfully build your components, you need to build the front end and package your application. The Maven `process-resources` goal generates the Node.js resources, creates the front-end production build, and copies and processes the resources into the destination directory. 
+After you successfully build your components, you need to build the front-end and package your application. The Maven `process-resources` goal generates the Node.js resources, creates the front-end production build, and copies and processes the resources into the destination directory.
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
After you successfully build your components, you need to build the front end and package your application. The Maven `process-resources` goal generates the Node.js resources, creates the front-end production build, and copies and processes the resources into the destination directory. 
```

File: README.adoc [hunk 20]
```diff
@@ -270,21 +266,21 @@ Add the [hotspot=axios-library]`axios` library and the [hotspot=get-posts]`GetAr
...
 // =================================================================================================
 
-== Building and packaging the front end
+== Building and packaging the front-end
 
-After you successfully build your components, you need to build the front end and package your application. The Maven `process-resources` goal generates the Node.js resources, creates the front-end production build, and copies and processes the resources into the destination directory. 
+After you successfully build your components, you need to build the front-end and package your application. The Maven `process-resources` goal generates the Node.js resources, creates the front-end production build, and copies and processes the resources into the destination directory. 
 
-In a new command-line session, build the front end by running the following command in the `start` directory:
+In a new command-line session, build the front-end by running the following command in the `start` directory:
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
In a new command-line session, build the front end by running the following command in the `start` directory:
```

File: README.adoc [hunk 21]
```diff
@@ -301,17 +297,16 @@ mvn process-resources
 ```
 endif::[]
 
-The build may take a few minutes to complete. You can rebuild the front end at any time with the Maven `process-resources` goal. Any local changes to your JavaScript and HTML are picked up when you build the front end.
+The build may take a few minutes to complete. You can rebuild the front-end at any time with the Maven `process-resources` goal. Any local changes to your JavaScript and HTML are picked up when you build the front-end.
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
The build may take a few minutes to complete. You can rebuild the front end at any time with the Maven `process-resources` goal. Any local changes to your JavaScript and HTML are picked up when you build the front-end.
```

File: README.adoc [hunk 22]
```diff
@@ -301,17 +297,16 @@ mvn process-resources
...
 
 ifndef::cloud-hosted[]
-Navigate to the http://localhost:9080[http://localhost:9080^] web application root to view the front end of your application.
+Navigate to the http://localhost:9080[http://localhost:9080^] web application root to view the front-end of your application.
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
Navigate to the http://localhost:9080[http://localhost:9080^] web application root to view the front end of your application.
```

File: README.adoc [hunk 23]
```diff
@@ -301,17 +297,16 @@ mvn process-resources
...
 endif::[]
 
 ifdef::cloud-hosted[]
-Open your browser and view the front end of your application by going to the URL that the following command returns:
-```bash
-echo http://${USERNAME}-9080.$(echo $TOOL_DOMAIN | sed 's/\.labs\./.proxy./g')
-```
+Click the following button to view the front-end of your application:
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
Click the following button to view the front end of your application:
```

File: README.adoc [hunk 24]
```diff
@@ -320,13 +315,12 @@ endif::[]
 
 == Testing the React client
 
-New projects that are created with `create-react-app` comes with a test file called [hotspot file]`App.test.js`, which is included in the `src/main/frontend/src` directory. The `App.test.js` file is a simple JavaScript file that tests against the `App.js` component. There are no explicit test cases that are written for this application. The `create-react-app` configuration uses https://jestjs.io/[Jest^] as its test runner.
-To learn more about Jest, go to their documentation on https://jestjs.io/docs/en/tutorial-react[Testing React apps^]. 
+`Next.js` supports a variety of testing tools. This guide uses `Vitest` for unit testing the React components, with the test file [hotspot file]`App.test.jsx` located in `src/main/frontend/__tests__/` directory. The `App.test.jsx` file is a simple JavaScript file that tests against the `page.jsx` component. There are no explicit test cases that are written for this application. To learn more about `Vitest`, please visit https://nextjs.org/docs/app/building-your-application/testing/vitest[Setting up Vitest with Next.js].
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
`Next.js` supports various testing tools. This guide uses `Vitest` for unit testing the React components, with the test file [hotspot file]`App.test.jsx` located in `src/main/frontend/__tests__/` directory. The `App.test.jsx` file is a simple JavaScript file that tests against the `page.jsx` component. No explicit test cases are written for this application. To learn more about `Vitest`, see https://nextjs.org/docs/app/building-your-application/testing/vitest[Setting up Vitest with Next.js].
```

File: README.adoc [hunk 25]
```diff
@@ -342,7 +336,33 @@ include::finish/pom.xml[]
 ----
 
 [role="edit_command_text"]
-To run the default test, you can add the [hotspot=node-tests file=1]`testing` configuration to the `frontend-maven-plugin`. Rerun the Maven `process-resources` goal to rebuild the front end and run the tests.
+To run the default test, you can add the [hotspot=node-tests file=1]`testing` configuration to the `frontend-maven-plugin`. Rerun the Maven `process-resources` goal to rebuild the front-end and run the tests.
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
To run the default test, you can add the [hotspot=node-tests file=1]`testing` configuration to the `frontend-maven-plugin`. Rerun the Maven `process-resources` goal to rebuild the front end and run the tests.
```

**Review by dmuelle — CHANGES_REQUESTED** [review]
ID review- left a some suggestions per IBM style.

GLobally- only hyphenate front end/ back end when they are used as adjectives ie

Build a front-end component
but
Build the front end of your application

Also, web page is 2 words. 

lmk if you have any questions-  thanks!

File: README.adoc [hunk 26]
```diff
@@ -35,15 +35,15 @@ You will learn how to access a REST service and deserialize the returned JSON th
...
 
-The React application in this guide is provided and configured for you in the `src/main/frontend` directory. The application uses the https://reactjs.org/docs/create-a-new-react-app.html[Create React App^] prebuilt configuration to set up the modern single-page React application. The https://github.com/facebook/create-react-app[create-react-app^] integrated toolchain is a comfortable environment for learning React and is the best way to start building a new single-page application with React.
+The React application in this guide is provided and configured for you in the `src/main/frontend` directory. The application uses https://nextjs.org/[Next.js^], a https://react.dev/learn/start-a-new-react-project[React-powered frameworks^], to set up the modern React application. The `Next.js` framework provides a powerful environment for learning and building React applications, with features like server-side rendering, static site generation, and easy API routes. It is the best way to start building a highly performant React application.
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
The React application in this guide is provided and configured for you in the `src/main/frontend` directory. The application uses https://nextjs.org/[Next.js^], a https://react.dev/learn/start-a-new-react-project[React-powered framework^], to set up the modern React application. The `Next.js` framework provides a powerful environment for learning and building React applications, with features like server-side rendering, static site generation, and easy API routes. It is the best way to start building a highly performant React application.
```

**Comment by Ruilin-Ma** [comment]
Hi @dmuelle, thanks for your review and suggestions. Appreciate that!

**Review by dmuelle — APPROVED** [review]
LGTM, thanks @Ruilin-Ma

**Review by gkwan-ibm — APPROVED** [review]

**Pull request merged**